library/std/src/sys/unix/thread.rs

```diff
@@ -112,8 +112,7 @@ impl Thread {
     }
 
     pub fn yield_now() {
-        let ret = unsafe { libc::sched_yield() };
-        debug_assert_eq!(ret, 0);
+        let _ = unsafe { libc::sched_yield() };
```

Comment:
Intentionally ignoring the return value here? If so, we should probably leave a comment about it.

Reply:
Still thinking that this makes sense, open a PR to have some review from @tavianator that looks like having some strong opinion on it.

I personally like an assertion here because if something bad happens I can find it, but in this case when something bad happens if is it useful to know this info?

Reply:
This function is documented to be infallible on Linux, but on BSDs it is documented to be able to fail with ENOSYS, if this system call is not enabled. This is a perfect place for a debug assertion.

Reply:
Reintrodiced!